### PR TITLE
[https://nvbugs/6566726][fix] Replan FlashInfer wrappers before CUDA graph replay

### PR DESCRIPTION
**Pull request by BoyueZ**
## Dev Engineer Review

- `FlashInferWrappers` now uses the shared `workspace_buffer` from metadata.
- Per-plan CUDA graph workspace allocation and ownership tracking were removed.
- Persistent wrappers are planned in `prepare()` for CUDA graph capture.
- Eager multi-wrapper planning remains deferred to `forward_impl()`.
- The changes preserve the intended workspace memory behavior and do not add public API or configuration changes.
- No configuration or test-list files changed.

## QA Engineer Review

- Test changes remove workspace-isolation and workspace-ownership tests.
- CUDA graph tests now verify persistent-wrapper planning before capture.
- Eager planning tests verify single-layer and multi-layer behavior.
- The changed test coverage includes CUDA graph planning, deferred eager planning, and replay output matching eager execution.
- Integration test-list coverage was not provided. Verdict: needs follow-up.

 
 ## Description
`FlashInferAttentionMetadata.prepare()` refreshes KV page metadata before execution. With multiple persistent wrappers, such as Gemma3 global and sliding-window attention, the existing code deferred replanning to `forward_impl()`.

This is valid for eager execution, but CUDA graph replay does not re-enter Python. As a result, replay could use a schedule generated for earlier KV page metadata.

This change disables deferred planning only for CUDA graph metadata. Persistent wrappers are replanned in `prepare()` before capture or replay, while eager multi-wrapper behavior remains unchanged.

FlashInfer wrappers continue to share the float runtime scratch workspace. Each wrapper retains its private integer plan workspace. This change adds no workspace allocation or memory overhead.

There are no public API, configuration, dependency, CODEOWNERS, documentation, or architecture changes.

## Test Coverage

- `pytest -q tests/unittest/_torch/attention/test_flashinfer_attention.py`
- Result: `15 passed`
- Verifies CUDA graph single- and multi-wrapper plans are ready after `prepare()`.
- Verifies eager multi-wrapper planning remains deferred.
- Verifies CUDA graph replay output matches eager execution.

  - Gemma3-27B FP8 with FlashInfer and CUDA graph enabled on B300/SM103:
  - Latest main: `44.6667%` (`402/900`)
  - After fix: `50.7778%` (`457/900`)
  - Note: On Blackwell, the baseline still passes the configured threshold, but the fix improves accuracy. On Rubin, the baseline does not meet the accuracy requirement, so this fix is required.
 
  ## PR Checklist

  Please review the following before submitting your PR:
  - PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
  - PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
  - Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
  - If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
  - Any new dependencies have been scanned for license and vulnerabilities
  - [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
  - Documentation updated as needed
  - Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
  - The reviewers assigned automatically/manually are appropriate for the PR.

  - [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: tensorrt_llm/_torch/attention_backend/flashinfer.py [hunk 1]
```diff
@@ -322,6 +341,34 @@ def needs_plan(self, plan_params: PlanParams) -> bool:
...
+            num_kv_heads=plan_params.num_kv_heads,
+            head_dim=plan_params.head_dim,
+            q_dtype=plan_params.q_dtype,
+            kv_dtype=plan_params.kv_dtype,
+            attention_mask_type=plan_params.attention_mask_type,
+            sm_scale=plan_params.sm_scale,
+            window_left=plan_params.window_left,
+            kv_pool_id=plan_params.kv_pool_id,
+            q_len_per_req=plan_params.q_len_per_req,
+        )
```
[the remark this answers is not in the document]
Comment (BoyueZ) [reply tensorrt_llm/_torch/attention_backend/flashinfer.py]: Solved with 09d99a35d6beba71ac771efbf4f64bcc8d9d0335

**Review by yuxianq — APPROVED** [review]
